### DISCUSSION TITLE
provenance hash

### DISCUSSION THREAD

**Discussion by JeanPaul20** [nftchef/art-engine#246] [Help + Q&A]
So I have been reading about the provenance hash, here in the discussion, and also followed the link in the read me file.
Seems to be important, so the NFT projects can prove to the community that the NFT images and metadata were pre-determined beforehand.
Now I have this _provenance.json, but what do I do with it?
Do I upload this somewhere on IPFS or just save the file on my computer?

**Comment by nftchef** [comment]
Generally speaking, it's up to the project to publish it somewhere.
At the very least, it should be included in the smart contract, either at the time of deployment if it is ready, or via a function that sets it before the art is revealed.

something like
```
string public PROVENANCE_HASH; // keccak256

    function setProvenance(string memory _provenance) external onlyOwner {
        PROVENANCE_HASH = _provenance;
    }


```
Some projects even go as far as making the set function only work once so it cannot be changed.

There is more info in this discussion if you haven't seen it yet.
https://github.com/nftchef/art-engine/discussions/65
